packages/matter.js/src/fabric/FabricManager.ts

```diff
@@ -69,9 +76,16 @@ export class FabricManager {
             throw new MatterFlowError(`Fabric with index ${fabricIndex} already exists.`);
         }
         this.#fabrics.set(fabricIndex, fabric);
-        fabric.addRemoveCallback(() => this.removeFabric(fabricIndex));
+        fabric.addRemoveCallback(async () => this.removeFabric(fabricIndex));
         fabric.setPersistCallback((isUpdate = true) => {
-            this.persistFabrics();
+            const persistResult = this.persistFabrics();
+            if (MaybePromise.is(persistResult)) {
```

Comment:
You can use MaybePromise.then() for this code pattern.  I removed it in a lot of places because it made our stacks too deep but in places like this I think it's fine to use.